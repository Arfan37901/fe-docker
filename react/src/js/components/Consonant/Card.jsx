import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Tooltip from './Tooltip';

const CARD_STYLE = {
    WIDE: '3:2',
    SQUARE: '1:1',
    FULL: 'full-card',
};

const Card = (props) => {
    const {
        id,
        title,
        label,
        description,
        ctaLink,
        ctaLabel,
        image,
        bannerDescription,
        bannerFontColor,
        bannerBackgroundColor,
        bannerIcon,
        secondaryLabelText,
        allowBookmarking,
        isBookmarked,
        cardSavedIco,
        cardUnsavedIco,
        onClick,
        saveBookmarkText,
        unsaveBookmarkText,
        cardStyle,
        videoURL,
    } = props;

    const handleClick = (clickEvt) => {
        clickEvt.stopPropagation();
        onClick(id);
    };

    const setClassName = () => {
        const res = ['consonant-card'];
        const style = cardStyle.toLowerCase().trim();

        if (style === CARD_STYLE.FULL) res.push('consonant-card_full');
        if (style === CARD_STYLE.SQUARE) res.push('consonant-card_square');

        return res.join(' ');
    };

    return (
        <div className={setClassName()} id={id}>
            <div
                className="consonant-card--img"
                style={{ backgroundImage: `url("${image}")` }}>
                {
                    bannerDescription &&
                    bannerFontColor &&
                    bannerBackgroundColor &&
                    <span
                        className="consonant-card--banner"
                        style={({
                            backgroundColor: bannerBackgroundColor,
                            color: bannerFontColor,
                        })}>
                        {
                            bannerIcon &&
                            <img
                                src={bannerIcon}
                                alt=""
                                loading="lazy"
                                width="8"
                                className="consonant-card--banner-icon" />
                        }
                        {bannerDescription}
                    </span>
                }
                { videoURL && <span className="consonant-card--video-ico" />}
            </div>
            <div className="consonant-card--inner">
                {
                    label &&
                    <span className="consonant-card--label">{label}</span>
                }
                <h2
                    className="consonant-card--title"
                    dangerouslySetInnerHTML={{ __html: title }} />
                {
                    ![CARD_STYLE.SQUARE, CARD_STYLE.FULL]
                        .some(el => el === cardStyle.toLowerCase().trim()) &&
                        <Fragment>
                            <p
                                className="consonant-card--text"
                                dangerouslySetInnerHTML={{ __html: description }} />
                            <div className="consonant-card--footer-wrapper">
                                <div className="consonant-card--footer-info">
                                    {
                                        allowBookmarking &&
                                        <button
                                            data-tooltip-wrapper
                                            type="button"
                                            className={
                                                isBookmarked ?
                                                    'consonant-card--footer-btn consonant-card--footer-btn_active' :
                                                    'consonant-card--footer-btn'
                                            }
                                            onClick={handleClick}>
                                            {(cardSavedIco && cardUnsavedIco) ?
                                                <img
                                                    src={
                                                        isBookmarked ?
                                                            cardSavedIco :
                                                            cardUnsavedIco
                                                    }
                                                    width="16"
                                                    alt=""
                                                    loading="lazy" /> :
                                                <span className="consonant-card--bookmark-ico" />
                                            }
                                            <Tooltip text={
                                                isBookmarked ? unsaveBookmarkText : saveBookmarkText
                                            } />
                                        </button>

                                    }
                                    {
                                        secondaryLabelText &&
                                        <span className="consonant-card--secondary-text">{secondaryLabelText}</span>
                                    }
                                </div>
                                <a
                                    href={ctaLink}
                                    target="_blank"
                                    className="consonant-card--btn">{ctaLabel}
                                </a>
                            </div>
                        </Fragment>
                }
            </div>
        </div>
    );
};

export default Card;

Card.propTypes = {
    id: PropTypes.string.isRequired,
    ctaLink: PropTypes.string.isRequired,
    ctaLabel: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    label: PropTypes.string,
    description: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    bannerDescription: PropTypes.string,
    bannerFontColor: PropTypes.string,
    bannerBackgroundColor: PropTypes.string,
    bannerIcon: PropTypes.string,
    secondaryLabelText: PropTypes.string,
    allowBookmarking: PropTypes.bool.isRequired,
    isBookmarked: PropTypes.bool.isRequired,
    cardSavedIco: PropTypes.string.isRequired,
    cardUnsavedIco: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
    saveBookmarkText: PropTypes.string,
    unsaveBookmarkText: PropTypes.string,
    cardStyle: PropTypes.string,
    videoURL: PropTypes.string,
};

Card.defaultProps = {
    bannerIcon: '',
    bannerDescription: '',
    bannerFontColor: '',
    bannerBackgroundColor: '',
    secondaryLabelText: '',
    label: '',
    saveBookmarkText: 'Save card',
    unsaveBookmarkText: 'Unsave card',
    cardStyle: '',
    videoURL: '',
};
