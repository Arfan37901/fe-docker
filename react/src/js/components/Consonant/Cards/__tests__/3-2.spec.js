import React from 'react';
import { screen, logDOM } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Card from '../3-2';

import { DEFAULT_PROPS_3_2 } from '../../Helpers/Testing/Constants/Card';

import makeSetup from '../../Helpers/Testing/Utils/Settings';

const setup = makeSetup(Card, DEFAULT_PROPS_3_2);

describe('Consonant/Card', () => {
    test('should renders banner correctly', () => {
        const {
            props: {
                overlays: {
                    banner: {
                        description: bannerDescription,
                        fontColor: bannerFontColor,
                        backgroundColor: bannerBackgroundColor,
                        icon: bannerIcon,
                    },
                },
            },
        } = setup();

        const bannerElement = screen.getByTestId('consonant-card--banner');
        const bannerIconElement = screen.getByTestId('consonant-card--banner-icon');

        expect(bannerElement).toHaveStyle({
            color: bannerFontColor,
            backgroundColor: bannerBackgroundColor,
        });
        expect(bannerElement).toHaveTextContent(bannerDescription);
        expect(bannerIconElement).toHaveAttribute('src', bannerIcon);
    });
    test('should renders with badge', () => {
        const {
            props: {
                overlays: {
                    label: {
                        description: someBadgeText,
                    },
                },
            },
        } = setup();

        const badgeElement = screen.queryByText(someBadgeText);

        expect(badgeElement).not.toBeNull();
    });

    test('should renders with label', () => {
        setup({
            contentArea: {
                dateDetailText: {
                    endTime: '2021-10-11T21:00:00.000Z',
                    startTime: '2021-10-11T21:00:00.000Z',
                },
            },
        });

        const labelElement = screen.queryByTestId('3-2-card--label');

        expect(labelElement).not.toBeNull();
    });

    test('should renders with detail text in label ', () => {
        setup({
            contentArea: {
                detailText: 'datail label',
                dateDetailText: {
                    startTime: undefined,
                },
            },
        });

        const labelElement = screen.queryByText('datail label');

        expect(labelElement).not.toBeNull();
    });

    test('A 3:2 card should be able to render a logo', () => {
        setup();
        const logoAltText = screen.getByAltText('logo-alt-text');
        expect(logoAltText).not.toBeNull();
    });
});
