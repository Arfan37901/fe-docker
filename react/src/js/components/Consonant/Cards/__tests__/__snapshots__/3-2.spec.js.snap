// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Consonant/Card Check snapshots should renders correctly with different card style 1`] = `
<div
  className="consonant-aspect-ratio-3-2-card"
  data-testid="consonant-card"
  id="6d6438c8-1"
>
  <div
    className="consonant-aspect-ratio-3-2-card--img"
    data-testid="consonant-card--img"
    style={
      Object {
        "backgroundImage": "url(\\"https://caas-chimera.s3-us-west-1.amazonaws.com/sample.jpg\\")",
      }
    }
  >
    <span
      className="consonant-aspect-ratio-3-2-card--banner"
      data-testid="consonant-card--banner"
      style={
        Object {
          "backgroundColor": "#CB6F10",
          "color": "#FFFFFF",
        }
      }
    >
      <img
        alt=""
        className="consonant-aspect-ratio-3-2-card--banner-icon"
        data-testid="consonant-card--banner-icon"
        loading="lazy"
        src="some-banner-icon"
        width="8"
      />
      Some Banner Description
    </span>
    
    <a
      className="consonant-aspect-ratio-3-2-card--video-ico"
      href="some-video-url"
      rel="noreferrer"
      tabIndex="0"
      target="_blank"
    >
      Open video link
    </a>
  </div>
  <div
    className="consonant-aspect-ratio-3-2-card--inner"
  >
    <span
      className="consonant-aspect-ratio-3-2-card--label"
    >
      Some label
    </span>
    <h2
      className="consonant-aspect-ratio-3-2-card--title"
    >
      Some Title 1
    </h2>
    <p
      className="consonant-aspect-ratio-3-2-card--text"
    >
      Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ charsDescription with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars
    </p>
    <div
      className="consonant-card-footer"
    >
      <div
        className="consonant-card-footer--row"
      />
    </div>
  </div>
</div>
`;

exports[`Consonant/Card Check snapshots should renders correctly with different card style 2`] = `
<div
  className="consonant-aspect-ratio-3-2-card"
  data-testid="consonant-card"
  id="6d6438c8-1"
>
  <div
    className="consonant-aspect-ratio-3-2-card--img"
    data-testid="consonant-card--img"
    style={
      Object {
        "backgroundImage": "url(\\"https://caas-chimera.s3-us-west-1.amazonaws.com/sample.jpg\\")",
      }
    }
  >
    <span
      className="consonant-aspect-ratio-3-2-card--banner"
      data-testid="consonant-card--banner"
      style={
        Object {
          "backgroundColor": "#CB6F10",
          "color": "#FFFFFF",
        }
      }
    >
      <img
        alt=""
        className="consonant-aspect-ratio-3-2-card--banner-icon"
        data-testid="consonant-card--banner-icon"
        loading="lazy"
        src="some-banner-icon"
        width="8"
      />
      Some Banner Description
    </span>
    
    <a
      className="consonant-aspect-ratio-3-2-card--video-ico"
      href="some-video-url"
      rel="noreferrer"
      tabIndex="0"
      target="_blank"
    >
      Open video link
    </a>
  </div>
  <div
    className="consonant-aspect-ratio-3-2-card--inner"
  >
    <span
      className="consonant-aspect-ratio-3-2-card--label"
    >
      Some label
    </span>
    <h2
      className="consonant-aspect-ratio-3-2-card--title"
    >
      Some Title 1
    </h2>
    <p
      className="consonant-aspect-ratio-3-2-card--text"
    >
      Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ charsDescription with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars
    </p>
    <div
      className="consonant-card-footer"
    >
      <div
        className="consonant-card-footer--row"
      />
    </div>
  </div>
</div>
`;

exports[`Consonant/Card Check snapshots should renders correctly with different card style 3`] = `
<div
  className="consonant-aspect-ratio-3-2-card"
  data-testid="consonant-card"
  id="6d6438c8-1"
>
  <div
    className="consonant-aspect-ratio-3-2-card--img"
    data-testid="consonant-card--img"
    style={
      Object {
        "backgroundImage": "url(\\"https://caas-chimera.s3-us-west-1.amazonaws.com/sample.jpg\\")",
      }
    }
  >
    <span
      className="consonant-aspect-ratio-3-2-card--banner"
      data-testid="consonant-card--banner"
      style={
        Object {
          "backgroundColor": "#CB6F10",
          "color": "#FFFFFF",
        }
      }
    >
      <img
        alt=""
        className="consonant-aspect-ratio-3-2-card--banner-icon"
        data-testid="consonant-card--banner-icon"
        loading="lazy"
        src="some-banner-icon"
        width="8"
      />
      Some Banner Description
    </span>
    
    <a
      className="consonant-aspect-ratio-3-2-card--video-ico"
      href="some-video-url"
      rel="noreferrer"
      tabIndex="0"
      target="_blank"
    >
      Open video link
    </a>
  </div>
  <div
    className="consonant-aspect-ratio-3-2-card--inner"
  >
    <span
      className="consonant-aspect-ratio-3-2-card--label"
    >
      Some label
    </span>
    <h2
      className="consonant-aspect-ratio-3-2-card--title"
    >
      Some Title 1
    </h2>
    <p
      className="consonant-aspect-ratio-3-2-card--text"
    >
      Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ charsDescription with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars
    </p>
    <div
      className="consonant-card-footer"
    >
      <div
        className="consonant-card-footer--row"
      />
    </div>
  </div>
</div>
`;

exports[`Consonant/Card Check snapshots should renders correctly with different card style 4`] = `
<div
  className="consonant-aspect-ratio-3-2-card"
  data-testid="consonant-card"
  id="6d6438c8-1"
>
  <div
    className="consonant-aspect-ratio-3-2-card--img"
    data-testid="consonant-card--img"
    style={
      Object {
        "backgroundImage": "url(\\"https://caas-chimera.s3-us-west-1.amazonaws.com/sample.jpg\\")",
      }
    }
  >
    <span
      className="consonant-aspect-ratio-3-2-card--banner"
      data-testid="consonant-card--banner"
      style={
        Object {
          "backgroundColor": "#CB6F10",
          "color": "#FFFFFF",
        }
      }
    >
      <img
        alt=""
        className="consonant-aspect-ratio-3-2-card--banner-icon"
        data-testid="consonant-card--banner-icon"
        loading="lazy"
        src="some-banner-icon"
        width="8"
      />
      Some Banner Description
    </span>
    
    <a
      className="consonant-aspect-ratio-3-2-card--video-ico"
      href="some-video-url"
      rel="noreferrer"
      tabIndex="0"
      target="_blank"
    >
      Open video link
    </a>
  </div>
  <div
    className="consonant-aspect-ratio-3-2-card--inner"
  >
    <span
      className="consonant-aspect-ratio-3-2-card--label"
    >
      Some label
    </span>
    <h2
      className="consonant-aspect-ratio-3-2-card--title"
    >
      Some Title 1
    </h2>
    <p
      className="consonant-aspect-ratio-3-2-card--text"
    >
      Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ charsDescription with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars
    </p>
    <div
      className="consonant-card-footer"
    >
      <div
        className="consonant-card-footer--row"
      />
    </div>
  </div>
</div>
`;

exports[`Consonant/Card Check snapshots should renders with bookmarking 1`] = `
<div
  className="consonant-aspect-ratio-3-2-card"
  data-testid="consonant-card"
  id="6d6438c8-1"
>
  <div
    className="consonant-aspect-ratio-3-2-card--img"
    data-testid="consonant-card--img"
    style={
      Object {
        "backgroundImage": "url(\\"https://caas-chimera.s3-us-west-1.amazonaws.com/sample.jpg\\")",
      }
    }
  >
    <span
      className="consonant-aspect-ratio-3-2-card--banner"
      data-testid="consonant-card--banner"
      style={
        Object {
          "backgroundColor": "#CB6F10",
          "color": "#FFFFFF",
        }
      }
    >
      <img
        alt=""
        className="consonant-aspect-ratio-3-2-card--banner-icon"
        data-testid="consonant-card--banner-icon"
        loading="lazy"
        src="some-banner-icon"
        width="8"
      />
      Some Banner Description
    </span>
    
    <a
      className="consonant-aspect-ratio-3-2-card--video-ico"
      href="some-video-url"
      rel="noreferrer"
      tabIndex="0"
      target="_blank"
    >
      Open video link
    </a>
  </div>
  <div
    className="consonant-aspect-ratio-3-2-card--inner"
  >
    <span
      className="consonant-aspect-ratio-3-2-card--label"
    >
      Some label
    </span>
    <h2
      className="consonant-aspect-ratio-3-2-card--title"
    >
      Some Title 1
    </h2>
    <p
      className="consonant-aspect-ratio-3-2-card--text"
    >
      Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ charsDescription with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars Description with 200+ chars
    </p>
    <div
      className="consonant-card-footer"
    >
      <div
        className="consonant-card-footer--row"
      />
    </div>
  </div>
</div>
`;

exports[`Consonant/Card Consonant/Tooltip Check snapshots should renders correctly 1`] = `
<span
  className="consonant-tooltip"
>
  Tooltip text
</span>
`;
